maker/src/cfd.rs

```diff
@@ -482,41 +485,60 @@ impl<O, T, W> Actor<O, T, W> {
     async fn handle_accept_settlement(&mut self, msg: AcceptSettlement) -> Result<()> {
         let AcceptSettlement { order_id } = msg;
 
-        match self
-            .settlement_actors
-            .send_async(&order_id, collab_settlement::Accepted)
-            .await
-        {
-            Ok(_) => Ok(()),
-            Err(NotConnected(e)) => {
-                self.executor
-                    .execute(order_id, |cfd| {
-                        Ok(cfd.fail_collaborative_settlement(anyhow!(e)))
-                    })
-                    .await?;
-
-                bail!("Accept failed: No settlement in progress for order {order_id}")
+        // TODO: Handle case when taker upgraded after opening and *can* use libp2p
```

Comment:
I am sure you are aware of this, but I still state it: Since the old code for the `taker` was removed this `TODO` has to be actioned before merging, otherwise the taker is unable to collab close previously opened positions after an upgrade!

Reply:
it has to be actioned before *releasing* - and I fully intend to do it, will spin-off a couple of the follow-up tickets to deal with it. 
Don't want to make this PR even bigger than it is :)